ssl/t1_lib.c

```diff
@@ -788,7 +792,8 @@ static const uint16_t tls_default_sigalg[] = {
     TLSEXT_SIGALG_ecdsa_sha1, /* SSL_PKEY_ECC */
     TLSEXT_SIGALG_gostr34102001_gostr3411, /* SSL_PKEY_GOST01 */
     TLSEXT_SIGALG_gostr34102012_256_gostr34112012_256, /* SSL_PKEY_GOST12_256 */
-    TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512 /* SSL_PKEY_GOST12_512 */
+    TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512, /* SSL_PKEY_GOST12_512 */
+    0 /* SSL_PKEY_ED25519 */
```

Comment:
The curve values have nothing to do with Ed25519 support. Ed25519 follows the TLS 1.3 negotiation mechanism where signing is purely determined by signature algorithms and curves/groups only affect (EC)DH. Plenty of clients today support X25519 without Ed25519 (including OpenSSL 1.1.0).

Re TLS 1.1 and below, we don't allow Ed25519 at those versions either for the same reason. It's rather silly without a signaling mechanism, and this is one less combination to worry about when thinking about downgrade possibilities, especially since you need TLS 1.2 to get a strong transcript hash.